src/UCommerce.Transactions.Payments.Braintree/BraintreePaymentMethodService.cs

```diff
@@ -90,14 +102,15 @@ public override Payment RequestPayment(PaymentRequest paymentRequest)
         /// <param name="payment">The payment to process.</param>
         public override void ProcessCallback(Payment payment)
         {
+            if (payment.PaymentStatus.PaymentStatusId != (int)PaymentStatusCode.PendingAuthorization)
+                return;
+
 			string paymentFormUrl = payment.PaymentMethod.DynamicProperty<string>().PaymentFormUrl;
 			string acceptUrl = payment.PaymentMethod.DynamicProperty<string>().AcceptUrl;
 			string declineUrl = payment.PaymentMethod.DynamicProperty<string>().DeclineUrl;
-			
-			if (payment.PaymentStatus.PaymentStatusId != (int)PaymentStatusCode.PendingAuthorization)
-                return;
 
-            Result<Transaction> result = GetBraintreeGateway(payment.PaymentMethod).TransparentRedirect.ConfirmTransaction(HttpContext.Current.Request.Url.Query);
+            var result = CreateBraintreeTransaction(payment);
+
             if (!result.IsSuccess())
                 HttpContext.Current.Response.Redirect(paymentFormUrl == "(auto)"
                     ? string.Format("/{0}/{1}/PaymentRequest.axd?errorMessage={2}", payment.PaymentMethod.PaymentMethodId, payment["paymentGuid"], result.Message.Replace('\n', ';'))

```

Comment:
I think brackets and separate var for the if a condition would make it bit more readable.

Reply:
My goal was to stay away from changing the integration too much.
On line 128 the opposite check is being used, I'd like to keep it the same.